lib/modules/manager/asdf/extract.spec.ts

```diff
@@ -0,0 +1,91 @@
+import { extractPackageFile } from '.';
+
+describe('modules/manager/asdf/extract', () => {
+  describe('extractPackageFile()', () => {
+    it('returns a result', () => {
+      const res = extractPackageFile('nodejs 16.16.0\n');
+      expect(res.deps).toEqual([
+        {
+          currentValue: 'nodejs 16.16.0',
+          datasource: 'github-tags',
+          depName: 'node',
+          packageName: 'nodejs/node',
+          versioning: 'node',
+        },
+      ]);
+    });
+
+    it('ignores lines with unsupported tooling', () => {
+      const res = extractPackageFile('yarn 1.22.5\n');
+      expect(res.deps).toEqual([]);
+    });
+
+    it('only captures the first version', () => {
+      const res = extractPackageFile('nodejs 16.16.0 16.15.1\n');
+      expect(res.deps).toEqual([
+        {
+          currentValue: 'nodejs 16.16.0',
+          datasource: 'github-tags',
+          depName: 'node',
+          packageName: 'nodejs/node',
+          versioning: 'node',
+        },
+      ]);
+    });
+
+    it('ignores supported tooling with ref: versioning', () => {
+      const res = extractPackageFile('nodejs ref:234abc4\n');
+      expect(res.deps).toEqual([]);
```

Comment:
```suggestion
      expect(res).toBeNull();
```

don't return empty deps

Reply:
It can be ok though, especially here where there's multiple types of tools we don't support. Returning null is like "this is not a package file". Returning empty deps is like saying "we didn't find any supported deps".

BTW it could perhaps be better to include all detected tools but have them with a skip reason already, so it's more clear to the user.

Reply:
yes, that's better, i just changed the return value to what all other manager do.

@sawilde Please add all found deps and add an appropriate `skipReason` for unsupported values.
https://github.com/renovatebot/renovate/blob/6a93997863e767c83eec89a4e1d807ef802c19ab/lib/types/skip-reason.ts#L1-L48

Reply:
done